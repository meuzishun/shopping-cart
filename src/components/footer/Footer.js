import React from 'react';

function Footer() {
  return (
    <div>
      <p>Copyright &copy; meuzishun 2022</p>
      {console.log('Footer has rendered...')}
    </div>
  );
}

export default Footer;
