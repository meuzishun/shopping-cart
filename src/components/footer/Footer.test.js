import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Footer from './Footer';

describe('Footer component', () => {
  afterEach(cleanup);

  it('renders', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('contains an element with the proper text', () => {
    render(<Footer />);
    expect(screen.getByText('Copyright Â© meuzishun 2022')).toBeTruthy();
  });
});
