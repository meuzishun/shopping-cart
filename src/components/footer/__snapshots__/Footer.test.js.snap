// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders 1`] = `
<DocumentFragment>
  <footer>
    <p>
      Copyright Â© meuzishun 2022
    </p>
  </footer>
</DocumentFragment>
`;
